mlrun/api/db/sqldb/db.py

```diff
@@ -3030,3 +3030,142 @@ def create_data_version(self, session, version):
+        self._upsert(session, [background_task_record])
+
+    def get_background_task(
+        self, session, name: str, project: str = None
+    ) -> schemas.BackgroundTask:
+        background_task_record = self._get_background_task_record(
+            session, name, project
+        )
+
+        return self._transform_background_task_record_to_schema(background_task_record)
+
+    @staticmethod
+    def _transform_background_task_record_to_schema(
+        background_task_record: BackgroundTask,
+    ) -> schemas.BackgroundTask:
+        timeout = background_task_record.timeout
+        if mlrun.mlconf.background_tasks_timeout_defaults.mode == "enabled":
+            timeout = (
+                timeout
+                if timeout
+                else mlrun.mlconf.background_tasks_timeout_defaults.default
+            )
+
+        return schemas.BackgroundTask(
+            metadata=schemas.BackgroundTaskMetadata(
+                name=background_task_record.name,
+                project=background_task_record.project,
+                created=background_task_record.created,
+                updated=background_task_record.updated,
+                timeout=timeout,
+            ),
+            spec=schemas.BackgroundTaskSpec(),
+            status=schemas.BackgroundTaskStatus(
+                state=background_task_record.state,
+            ),
+        )
+
+    def _get_background_task_record(
+        self,
+        session: Session,
+        name: str,
+        project: str = None,
+        raise_on_not_found: bool = True,
+    ) -> BackgroundTask:
+        background_task_record = (
+            self._query(session, BackgroundTask, name=name)
+            .filter(BackgroundTask.project == project)
+            .one_or_none()
+        )
+        if not background_task_record:
+            if not raise_on_not_found:
+                return None
+            if project:
+                raise mlrun.errors.MLRunNotFoundError(
+                    f"Background task not found: name={name}, project={project}"
+                )
+            else:
+                raise mlrun.errors.MLRunNotFoundError(
+                    f"Background task not found: name={name}"
+                )
+        if self._is_background_task_timeout_exceeded(background_task_record):
+            # lazy update of state, only if get background task was requested and the timeout for the update passed
+            # and the task still in progress then we change to failed
+            self.store_background_task(
+                session,
+                name,
+                mlrun.api.schemas.background_task.BackgroundTaskState.failed,
+                project,
+            )
+            return self._get_background_task_record(session, name, project)
+
+        return background_task_record
+
+    @staticmethod
+    def _is_background_task_timeout_exceeded(background_task_record) -> bool:
+        if mlrun.mlconf.background_tasks_timeout_defaults.mode == "enabled":
```

Comment:
I again think we need to keep things simple - if some record has a timeout set apply it, if we'll want to disable timeouts, we'll do it, and from then on, timeout won't be set on records, and therefore later won't be apply on gets